src/models/hooks/activityReportGoal.js

```diff
@@ -152,6 +153,46 @@ const destroyLinkedSimilarityGroups = async (sequelize, instance, options) => {
   });
 };
 
+const updateOnARAndOnApprovedARForMergedGoals = async (sequelize, instance) => {
+  const changed = instance.changed();
+
+  // Check if both originalGoalId and goalId have been changed and originalGoalId is not null
+  if (Array.isArray(changed)
+    && changed.includes('originalGoalId')
+    && changed.includes('goalId')
+    && instance.originalGoalId !== null) {
+    const { goalId } = instance;
+
+    // Check if the ActivityReport linked to this ActivityReportGoal has a
+    // calculatedStatus of 'approved'
+    const approvedActivityReports = await sequelize.models.ActivityReport.count({
+      where: {
+        calculatedStatus: 'approved',
+        id: instance.activityReportId, // Use the activityReportId from the instance
+      },
+    });
+
```

Comment:
we don't want to update onAR if they aren't onApprovedAR?

Reply:
They can be on an AR that is not yet approved, can't they?

Reply:
ah I see that I missed an important detail; this is the hook for activityReportGoal not goal... my bad